src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1308,20 +1308,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
             const auto shiftEnabled = WI_IsFlagSet(modifiers, static_cast<uint32_t>(VirtualKeyModifiers::Shift));
             const auto ctrlEnabled = WI_IsFlagSet(modifiers, static_cast<uint32_t>(VirtualKeyModifiers::Control));
 
-            if (_CanSendVTMouseInput())
+            auto lock = _terminal->LockForWriting();
+            const auto cursorPosition = point.Position();
+            const auto terminalPosition = _GetTerminalPosition(cursorPosition);
+
+            // GH#9396: we prioritize hyper-link over VT mouse events
+            if (point.Properties().IsLeftButtonPressed() && ctrlEnabled && !_terminal->GetHyperlinkAtPosition(terminalPosition).empty())
+            {
+                // Handle hyper-link only on the first click to prevent multiple activations
+                auto clickCount = _NumberOfClicks(cursorPosition, point.Timestamp());

```

Comment:
```suggestion
                const auto clickCount = _NumberOfClicks(cursorPosition, point.Timestamp());
```

Reply:
Also, could we just move this outside of the if-statement over by line 1313? That way we don't do it again on line 1335?

Reply:
My only concern with that would be that we're updating the click count whenever we try to send VT Mouse Input, but I think that's ok. Mind playing with that a little bit to make sure though?

Reply:
@carlos-zamora - yes, I didn't put the clicks count earlier to avoid counting the number of clicks in VT mouse mode. I am not sure how important it is, but Dustin was somewhat worried with that in my original solution.

Reply:
Fair enough. Can I at least get a `const` on the two `clickCount`s then? :P

Reply:
> 
> 
> Fair enough. Can I at least get a `const` on the two `clickCount`s then? :P

Lol! Absolutely! Doing it :)

Reply:
@carlos-zamora - applied the consts. Thanks for the feedback!